Gsit has a very serious problem running on Folia

Gsit keeps shutting down the server inexplicably.

How to reproduce it I don't know, but the reported errors are all from Gsit.

I've tried converting gsit's storage method sqlite to Mysql, but it still gets an automatic server shutdown by gsit.

```
`[20:49:40] [Region Scheduler Thread #7/ERROR]: [ca.spottedleaf.moonrise.common.util.TickThread] Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #7,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-503, -1],world=world}], entity={root=[{type=ServerPlayer,id=22385462,uuid=a0d6f579-1060-454a-91ba-d64eb0423b4d,pos=(-7,898.500,69.000,-82.623),mot=(0.000,0.000,0.000),aabb=AABB[-3740.317676582507, 107.5, 5559.946756709954] -> [-3739.717676558665, 109.29999995231628, 5560.5467567337955],removed=null,has_vehicle=false,passenger_count=0], vehicle=[{null}], passengers=[]
java.lang.Throwable: null
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:97) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:44) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:528) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.moveTo(Entity.java:2104) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at GSit-2.3.2.jar/dev.geco.gsit.mcv.v1_21_4.util.EntityUtil.setEntityLocation(EntityUtil.java:46) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.SitService.lambda$handleSafeSeatDismount$3(SitService.java:171) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.TaskService.lambda$runDelayed$5(TaskService.java:104) ~[GSit-2.3.2.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$LocationScheduledTask.run(FoliaRegionScheduler.java:331) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$Scheduler.tick(FoliaRegionScheduler.java:272) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler.tick(FoliaRegionScheduler.java:142) ~[folia-1.21.4.jar:?]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1646) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[20:49:40] [Region Scheduler Thread #3/ERROR]: [ca.spottedleaf.moonrise.common.util.TickThread] Thread failed main thread check: Cannot getEntities asynchronously, context=[thread=Region Scheduler Thread #3,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-226, 353],world=world}], world=world, aabb=AABB[-7898.800000011921, 69.0, -82.92300001192093] -> [-7898.199999988079, 70.79999995231628, -82.32299998807908]
java.lang.Throwable: null
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:106) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.level.Level.getEntities(Level.java:1807) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.pushEntities(LivingEntity.java:3612) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.pushEntities(ServerPlayer.java:2206) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3528) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:355) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.tick(Connection.java:673) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:521) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[20:49:40] [Region Scheduler Thread #7/WARN]: [GSit] Location task for GSit v2.3.2 in world CraftWorld{name=world} at -494, -6 generated an exception
java.lang.IllegalStateException: Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #7,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-503, -1],world=world}], entity={root=[{type=ServerPlayer,id=22385462,uuid=a0d6f579-1060-454a-91ba-d64eb0423b4d,pos=(-7,898.500,69.000,-82.623),mot=(0.000,0.000,0.000),aabb=AABB[-3740.317676582507, 107.5, 5559.946756709954] -> [-3739.717676558665, 109.29999995231628, 5560.5467567337955],removed=null,has_vehicle=false,passenger_count=0], vehicle=[{null}], passengers=[]
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:98) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:44) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:528) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.moveTo(Entity.java:2104) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at GSit-2.3.2.jar/dev.geco.gsit.mcv.v1_21_4.util.EntityUtil.setEntityLocation(EntityUtil.java:46) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.SitService.lambda$handleSafeSeatDismount$3(SitService.java:171) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.TaskService.lambda$runDelayed$5(TaskService.java:104) ~[GSit-2.3.2.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$LocationScheduledTask.run(FoliaRegionScheduler.java:331) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$Scheduler.tick(FoliaRegionScheduler.java:272) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler.tick(FoliaRegionScheduler.java:142) ~[folia-1.21.4.jar:?]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1646) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[20:49:40] [Region Scheduler Thread #3/WARN]: [io.papermc.paper.threadedregions.RegionizedWorldData] Failed to handle packet for /111.60.239.72:59076
net.minecraft.ReportedException: Ticking player
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1123) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:355) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.tick(Connection.java:673) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:521) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.IllegalStateException: Thread failed main thread check: Cannot getEntities asynchronously, context=[thread=Region Scheduler Thread #3,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-226, 353],world=world}], world=world, aabb=AABB[-7898.800000011921, 69.0, -82.92300001192093] -> [-7898.199999988079, 70.79999995231628, -82.32299998807908]
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:107) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.level.Level.getEntities(Level.java:1807) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.pushEntities(LivingEntity.java:3612) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.pushEntities(ServerPlayer.java:2206) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3528) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	... 9 more
[20:49:40] [Region Scheduler Thread #3/INFO]: Mioyyyyy lost connection: Internal server error
[20:49:40] [Region Scheduler Thread #3/ERROR]: [ca.spottedleaf.moonrise.common.util.TickThread] Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #3,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-226, 353],world=world}], world=world, chunk_pos=[-494, -6]
java.lang.Throwable: null
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:88) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:39) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:543) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPos(Entity.java:795) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.move(Entity.java:1157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.handleRelativeFrictionAndCalculateMovement(LivingEntity.java:3067) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travelInAir(LivingEntity.java:2905) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travel(LivingEntity.java:2896) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.travel(Player.java:1637) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3496) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:609) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:593) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.removePlayerFromWorld(ServerGamePacketListenerImpl.java:2157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2134) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2120) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.handleDisconnection(Connection.java:933) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:508) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[20:49:40] [Region Scheduler Thread #3/ERROR]: [io.papermc.paper.threadedregions.TickRegionScheduler] Region #3085443 centered at chunk [-226, 353] in world 'world' failed to tick:
net.minecraft.ReportedException: Ticking player
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1123) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:609) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:593) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.removePlayerFromWorld(ServerGamePacketListenerImpl.java:2157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2134) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2120) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.handleDisconnection(Connection.java:933) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:508) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.IllegalStateException: Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #3,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-226, 353],world=world}], world=world, chunk_pos=[-494, -6]
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:89) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:39) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:543) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPos(Entity.java:795) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.move(Entity.java:1157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.handleRelativeFrictionAndCalculateMovement(LivingEntity.java:3067) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travelInAir(LivingEntity.java:2905) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travel(LivingEntity.java:2896) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.travel(Player.java:1637) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3496) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	... 13 more
[20:49:40] [Region shutdown thread/INFO]: [RegionShutdownThread] Awaiting scheduler termination for 60s...
[20:49:40] [Region shutdown thread/INFO]: [RegionShutdownThread] Scheduler halted
[20:49:40] [Region shutdown thread/INFO]: Stopping server
[20:49:40] [Region shutdown thread/INFO]: [AntiPopup] Disabling AntiPopup v10.1`
```

Re-occurrence of this error causes the server to shut down, which happens almost every day.

```
[00:07:09] [Region Scheduler Thread #4/ERROR]: [ca.spottedleaf.moonrise.common.util.TickThread] Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #4,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-503, -3],world=world}], entity={root=[{type=ServerPlayer,id=16881506,uuid=1177a443-82ee-46d7-871a-ba3f2ca826c3,pos=(-7,952.500,81.062,-57.500),mot=(0.000,0.000,0.000),aabb=AABB[-4207.461251821958, 63.0, 5748.366507356696] -> [-4206.861251798116, 64.79999995231628, 5748.966507380538],removed=null,has_vehicle=false,passenger_count=0], vehicle=[{null}], passengers=[]
java.lang.Throwable: null
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:97) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:44) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:528) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.moveTo(Entity.java:2104) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at GSit-2.3.2.jar/dev.geco.gsit.mcv.v1_21_4.util.EntityUtil.setEntityLocation(EntityUtil.java:46) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.SitService.lambda$handleSafeSeatDismount$3(SitService.java:171) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.TaskService.lambda$runDelayed$5(TaskService.java:104) ~[GSit-2.3.2.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$LocationScheduledTask.run(FoliaRegionScheduler.java:331) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$Scheduler.tick(FoliaRegionScheduler.java:272) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler.tick(FoliaRegionScheduler.java:142) ~[folia-1.21.4.jar:?]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1646) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[00:07:09] [Region Scheduler Thread #4/WARN]: [GSit] Location task for GSit v2.3.2 in world CraftWorld{name=world} at -498, -4 generated an exception
java.lang.IllegalStateException: Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #4,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-503, -3],world=world}], entity={root=[{type=ServerPlayer,id=16881506,uuid=1177a443-82ee-46d7-871a-ba3f2ca826c3,pos=(-7,952.500,81.062,-57.500),mot=(0.000,0.000,0.000),aabb=AABB[-4207.461251821958, 63.0, 5748.366507356696] -> [-4206.861251798116, 64.79999995231628, 5748.966507380538],removed=null,has_vehicle=false,passenger_count=0], vehicle=[{null}], passengers=[]
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:98) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:44) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:528) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.moveTo(Entity.java:2104) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at GSit-2.3.2.jar/dev.geco.gsit.mcv.v1_21_4.util.EntityUtil.setEntityLocation(EntityUtil.java:46) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.SitService.lambda$handleSafeSeatDismount$3(SitService.java:171) ~[GSit-2.3.2.jar:?]
	at GSit-2.3.2.jar/dev.geco.gsit.service.TaskService.lambda$runDelayed$5(TaskService.java:104) ~[GSit-2.3.2.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$LocationScheduledTask.run(FoliaRegionScheduler.java:331) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler$Scheduler.tick(FoliaRegionScheduler.java:272) ~[folia-1.21.4.jar:?]
	at io.papermc.paper.threadedregions.scheduler.FoliaRegionScheduler.tick(FoliaRegionScheduler.java:142) ~[folia-1.21.4.jar:?]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1646) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[00:07:09] [Region Scheduler Thread #0/ERROR]: [ca.spottedleaf.moonrise.common.util.TickThread] Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #0,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-260, 349],world=world}], world=world, chunk_pos=[-498, -4]
java.lang.Throwable: null
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:88) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:39) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:543) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPos(Entity.java:795) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.move(Entity.java:1157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.handleRelativeFrictionAndCalculateMovement(LivingEntity.java:3067) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travelInAir(LivingEntity.java:2905) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travel(LivingEntity.java:2896) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.travel(Player.java:1637) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3496) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:355) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.tick(Connection.java:673) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:521) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[00:07:09] [Region Scheduler Thread #0/WARN]: [io.papermc.paper.threadedregions.RegionizedWorldData] Failed to handle packet for /27.15.154.47:56591
net.minecraft.ReportedException: Ticking player
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1123) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:355) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.tick(Connection.java:673) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:521) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.IllegalStateException: Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #0,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-260, 349],world=world}], world=world, chunk_pos=[-498, -4]
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:89) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:39) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:543) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPos(Entity.java:795) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.move(Entity.java:1157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.handleRelativeFrictionAndCalculateMovement(LivingEntity.java:3067) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travelInAir(LivingEntity.java:2905) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travel(LivingEntity.java:2896) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.travel(Player.java:1637) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3496) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	... 9 more
[00:07:09] [Region Scheduler Thread #0/INFO]: Ender_Ycer lost connection: Internal server error
[00:07:09] [Region Scheduler Thread #0/ERROR]: [ca.spottedleaf.moonrise.common.util.TickThread] Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #0,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-260, 349],world=world}], world=world, chunk_pos=[-498, -4]
java.lang.Throwable: null
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:88) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:39) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:543) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPos(Entity.java:795) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.move(Entity.java:1157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.handleRelativeFrictionAndCalculateMovement(LivingEntity.java:3067) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travelInAir(LivingEntity.java:2905) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travel(LivingEntity.java:2896) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.travel(Player.java:1637) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3496) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:609) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:593) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.removePlayerFromWorld(ServerGamePacketListenerImpl.java:2157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2134) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2120) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.handleDisconnection(Connection.java:933) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:508) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[00:07:09] [Region Scheduler Thread #0/ERROR]: [io.papermc.paper.threadedregions.TickRegionScheduler] Region #2235086 centered at chunk [-260, 349] in world 'world' failed to tick:
net.minecraft.ReportedException: Ticking player
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1123) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:609) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.players.PlayerList.remove(PlayerList.java:593) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.removePlayerFromWorld(ServerGamePacketListenerImpl.java:2157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2134) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.onDisconnect(ServerGamePacketListenerImpl.java:2120) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.network.Connection.handleDisconnection(Connection.java:933) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.RegionizedWorldData.tickConnections(RegionizedWorldData.java:508) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1849) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1668) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegions$ConcreteRegionTickHandle.tickRegion(TickRegions.java:407) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:418) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:546) ~[concurrentutil-0.0.3.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.IllegalStateException: Thread failed main thread check: Cannot move entity off-main, context=[thread=Region Scheduler Thread #0,class=io.papermc.paper.threadedregions.TickRegionScheduler$TickThreadRunner,region={center=[-260, 349],world=world}], world=world, chunk_pos=[-498, -4]
	at ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread(TickThread.java:89) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.server.ServerEntityLookup.checkThread(ServerEntityLookup.java:39) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup.moveEntity(EntityLookup.java:543) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at ca.spottedleaf.moonrise.patches.chunk_system.level.entity.EntityLookup$EntityCallback.onMove(EntityLookup.java:978) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPosRaw(Entity.java:5633) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.setPos(Entity.java:795) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.Entity.move(Entity.java:1157) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.handleRelativeFrictionAndCalculateMovement(LivingEntity.java:3067) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travelInAir(LivingEntity.java:2905) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.travel(LivingEntity.java:2896) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.travel(Player.java:1637) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3496) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.aiStep(Player.java:591) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3179) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.world.entity.player.Player.tick(Player.java:298) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:1041) ~[folia-1.21.4.jar:1.21.4-DEV-cd49d65]
	... 13 more
[00:07:09] [Region shutdown thread/INFO]: [RegionShutdownThread] Awaiting scheduler termination for 60s...
[00:07:09] [Region shutdown thread/INFO]: [RegionShutdownThread] Scheduler halted
[00:07:09] [Region shutdown thread/INFO]: Stopping server
[00:07:09] [Region shutdown thread/INFO]: [AntiPopup] Disabling AntiPopup v10.1
[00:07:09] [Region shutdown thread/INFO]: [AntiPopup] Disabled PacketEvents.
```

--- a/core/src/main/java/dev/geco/gsit/service/SitService.java
+++ b/core/src/main/java/dev/geco/gsit/service/SitService.java
@@ -170,7 +170,7 @@ public class SitService {
             if(seat.getSeatEntity().isValid()) gSitMain.getEntityUtil().setEntityLocation(seat.getSeatEntity(), returnLocation);
             if(gSitMain.supportsFoliaFeature()) gSitMain.getTaskService().runDelayed(() -> {
                 if(entity.isValid()) gSitMain.getEntityUtil().setEntityLocation(entity, returnLocation);
-            }, entity, 0);
+            }, true, Bukkit.isOwnedByCurrentRegion(entity) ? entity : null, 0);
             else if(entity.isValid()) gSitMain.getEntityUtil().setEntityLocation(entity, returnLocation);
 
             entityBlocked.remove(entity.getUniqueId());

